fslock.go

```diff
@@ -35,30 +59,20 @@ func Locked(confdir, lockFile string) (bool, error) {
 
 	lk, err := Lock(confdir, lockFile)
 	if err != nil {
```

Comment:
Given that we're now parsing the error in `Lock`, what if we just had `Lock` return a custom error type on failure? We could return an error implementing:

```go
type myerror interface {
  Permission() bool // returns true if this is a permission error.
  Locked() bool // returns true if this error means that the lock has already been taken.
}
```

Then we can provide two functions to determine why taking the lock failed:

```go
func IsLocked(err) bool { ... }
func IsPermission(err) bool { ... }
```

_This_ function would be reduced to an `IsLocked(err)` call (and a debug statement).

Reply:
@Stebalien I have been thinking about this and I have to learn more about idiomatic error handling in Go before I make any changes.

On the other hand, I'm not sure that the [original issue](https://github.com/ipfs/go-ipfs/issues/6434) has been solved by my changes.

What do you think?

Reply:
The original issue has been solved, I'm just trying to avoid parsing errors in two places.